### PR TITLE
Compare screenshot images from two commit

### PR DESCRIPTION
**Pull request by shifucun**
Example diff (with minor source order change)

image

### CODE REVIEW

File: server/routes/screenshot/html.py [hunk 1]
```diff
@@ -12,22 +12,61 @@
...
+@bp.route('/compare/<path:compare>')
+def diff(compare):
+  if os.environ.get('FLASK_ENV') not in [
+      'autopush', 'local', 'test', 'webdriver'
+  ]:
+    flask.abort(404)
+
+  parts = compare.split('...')
+  if len(parts) != 2:
+    return "Invalid hash comparison: " + compare, 400
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply server/routes/screenshot/html.py:53]: fix this by escaping `compare`?

**Review by chejennifer — APPROVED** [review]
this is really cool!!

**Review by chejennifer — APPROVED** [review]
this is really cool!!

File: server/routes/screenshot/html.py [hunk 2]
```diff
@@ -12,22 +12,61 @@
...
+  data = {}
+  for name in images:
+    data[name] = {
+        'base': b64encode(images[name]).decode('utf-8'),
+    }
+  return flask.render_template('screenshot.html', data=data, githash=githash)
+
+
+@bp.route('/compare/<path:compare>')
+def diff(compare):
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/routes/screenshot/html.py:46]: Done

File: server/routes/screenshot/html.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shifucun) [reply server/routes/screenshot/html.py:60]: Done

**Review by shifucun — COMMENTED** [review]
thanks!

**Pull request merged**